packages/preview/web/src/Interactions/interactionMap.tsx

```diff
@@ -1,22 +1,24 @@
-import React from 'react';
+import React, { ComponentType } from 'react';
 import { INTERACTIONS } from '../const';
 import ReadMore from '../Components/ReadMore';
 import SeeFullPost from '../Components/SeeFullPost';
 import ImageCounter from '../Components/ImageCounter';
+import { PreviewConfig } from '..';
 
 /* eslint-disable react/prop-types */
-export const interactionMap = onPreviewExpand =>
+interface GenericInteractionProps extends Record<string, unknown> {
+  onPreviewExpand: PreviewConfig['onPreviewExpand'];
+}
+export const interactionMap = (onPreviewExpand: PreviewConfig['onPreviewExpand']) =>
   Object.entries({
-    [INTERACTIONS.READ_MORE]: ReadMore,
-    [INTERACTIONS.SEE_FULL_CONTENT]: SeeFullPost,
-    [INTERACTIONS.IMAGE_COUNTER]: ImageCounter,
+    [INTERACTIONS.READ_MORE]: ReadMore as ComponentType,
+    [INTERACTIONS.SEE_FULL_CONTENT]: SeeFullPost as ComponentType,
+    [INTERACTIONS.IMAGE_COUNTER]: ImageCounter as ComponentType,
   }).reduce(
     (map, [key, Component]) => ({
```

Comment:
```suggestion
    (map, [key, Component]: [string, ComponentType]) => ({
```
Can be used instead of `as ComponentType` above

Reply:
yup, cool idea